backend/lib/io.js

```diff
@@ -72,75 +75,72 @@ function leaveCommentsRooms (socket) {
   leaveRooms(socket, room => room !== 'issues')
 }
 
-async function subscribeShoots ({ socket, namespacesAndFilters, projectList }) {
+async function subscribeShoots (socket, { namespace, namespaces, filter, user }) {
   leaveShootsAndShootRoom(socket)
 
-  /* join current rooms */
-  if (!_.isArray(namespacesAndFilters)) {
-    return
+  // subscribe for all namespaces or a single namespace
+  if (Array.isArray(namespaces)) {
+    namespace = '_all'
+  } else if (namespace) {
+    namespaces = [namespace]
+  } else {
+    assert.fail('Either property namespaces or namespace is required')
```

Comment:
All errors thrown by the `assert` module are instances of the [`AssertionError`](https://nodejs.org/api/assert.html#class-assertassertionerror) class. The reason why I have used an assertion here is that this assertion should never ever fail. Currently this method is called from the following two shoot different shoot subscriptions:

**Subscribe shoots for all namespaces the user is authorized**
```js
  socket.on('subscribeAllShoots', async ({ filter }) => {
    try {
      const user = getUserFromSocket(socket)
      const projectList = await projects.list({ user })
      const namespaces = _.map(projectList, 'metadata.namespace')
      
      if (await authorization.isAdmin(user)) {
        subscribeShootsAdmin(socket, { namespaces, filter, user })
      } else {
        subscribeShoots(socket, { namespaces, filter, user })
      }
    } catch (err) {
      ...
    }
  })
```
In this case the `namespaces` are passed to the `subscribeShoots` function.

**Subscribe shoots for a single namespace**
```js
  socket.on('subscribeShoots', async ({ namespace, filter }) => {
    ...
    try {
      const user = getUserFromSocket(socket)
      ...
      subscribeShoots(socket, { namespace, filter, user })
    } catch (err) {
      ...
    }
  })
```
In this case the `namespace` is passed to the `subscribeShoots` function.

It is important that we do not await the result of call to `subscribeShoots` in both cases here. This means the surrounding try catch block does not catch the `AssertionError`. But this is the intended behavior of this assertion. If for whatever reasons the assertion fails the developer should immediately be informed about the fact that the assertions is not met anymore. The unit tests will never be successful with a failing assertion here because these lines are covered by our tests.